src/NHibernate/Impl/SessionIdLoggingContext.cs

```diff
@@ -8,10 +8,12 @@ public class SessionIdLoggingContext : IDisposable
 		private static readonly AsyncLocal<Guid?> _currentSessionId = new AsyncLocal<Guid?>();
 
 		private readonly Guid? _oldSessonId;
+		private static bool _tracking;
 
-		public SessionIdLoggingContext(Guid id)
+		public SessionIdLoggingContext(Guid id, bool track)
 		{
-			if (!Enabled)
+			_tracking = track;
+			if (!_tracking)
```

Comment:
I would have just tested `id == Guid.Empty` instead of adding the tracking flag.

Reply:
The tracking flag is needed for the `Dispose` method, otherwise we have to check the `SessionId` that is slow due to `AsyncLocal`.

Reply:
Yes for the one in SessionIdLoggingContext, I should have been more precise and write `track`, the one coming from the added boolean on `ISessionImplementor`.